From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: anhcraft <huynhduyanh123123@gmail.com>
Date: Thu, 1 Jun 2023 08:13:21 +0700
Subject: [PATCH] Backport: Optimize entity tracker passenger checks


diff --git a/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java
index 67ca28463f5add7c18f7f16b918c3f36f8feeeda..c8bc59a73b560f1a468a785a49fa243bbbf4ce8d 100644
--- a/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java
+++ b/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java
@@ -88,7 +88,7 @@ public class EntityTrackerEntry {
         this.trackedPlayers = trackedPlayers;
         // CraftBukkit end
         this.m = Vec3D.ORIGIN;
-        this.p = Collections.emptyList();
+        this.p = com.google.common.collect.ImmutableList.of();
         this.b = worldserver;
         this.f = consumer;
         this.tracker = entity;
