From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: anhcraft <huynhduyanh123123@gmail.com>
Date: Thu, 1 Jun 2023 09:49:49 +0700
Subject: [PATCH] Backport: Fix command suggestion leak


diff --git a/src/main/java/com/mojang/brigadier/CommandDispatcher.java b/src/main/java/com/mojang/brigadier/CommandDispatcher.java
index e8fdbe7b8d8192a3247d98534e78ede7a7314a91..8cefd8cc6200f6d81056e48ea90000c9d7b9bcee 100644
--- a/src/main/java/com/mojang/brigadier/CommandDispatcher.java
+++ b/src/main/java/com/mojang/brigadier/CommandDispatcher.java
@@ -593,10 +593,14 @@ public class CommandDispatcher<S> {
         final String remainingLower = truncatedInput.substring(start).toLowerCase(); // Tuinity
         for (final CommandNode<S> node : parent.getChildren()) {
             CompletableFuture<Suggestions> future = Suggestions.empty();
+            // Paper start - Don't suggest if the requirement isn't met
+            if (parent != this.root || node.canUse(context.getSource())) {
             try {
                 future = node.listSuggestions(context.build(truncatedInput), new SuggestionsBuilder(truncatedInput, start, remainingLower)); // Tuinity
             } catch (final CommandSyntaxException ignored) {
             }
+            }
+            // Paper end
             futures[i++] = future;
         }
 
