From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: anhcraft <huynhduyanh123123@gmail.com>
Date: Thu, 1 Jun 2023 10:00:57 +0700
Subject: [PATCH] Backport: Do not use ForkJoinPool common exector for creating
 client command struct


diff --git a/src/main/java/net/minecraft/commands/CommandDispatcher.java b/src/main/java/net/minecraft/commands/CommandDispatcher.java
index 077beec1c1f0ae99854fe9e57fccfaaa94bbfd33..62ff0bb458cb155a6465c3bbddd834dfe25dc13b 100644
--- a/src/main/java/net/minecraft/commands/CommandDispatcher.java
+++ b/src/main/java/net/minecraft/commands/CommandDispatcher.java
@@ -344,9 +344,7 @@ public class CommandDispatcher {
         // CraftBukkit start
         // Register Vanilla commands into builtRoot as before
         // Paper start - Async command map building
-        java.util.concurrent.ForkJoinPool.commonPool().execute(() -> {
-            sendAsync(entityplayer);
-        });
+        net.minecraft.server.MCUtil.scheduleAsyncTask(() -> this.sendAsync(entityplayer));
     }
 
     private void sendAsync(EntityPlayer entityplayer) {
