From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: anhcraft <huynhduyanh123123@gmail.com>
Date: Thu, 1 Jun 2023 10:11:27 +0700
Subject: [PATCH] Backport: Fix NPE when getting destroy speed of air


diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
index 0006b3cad5fe46e50b0efeae99102f9d80276d61..58ed3345a961ad0cc184ebeb13d32cfa40ff42fc 100644
--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
@@ -794,6 +794,9 @@ public class CraftBlock implements Block {
         net.minecraft.world.item.ItemStack nmsItemStack;
         if (itemStack instanceof CraftItemStack) {
             nmsItemStack = ((CraftItemStack) itemStack).getHandle();
+            if (nmsItemStack == null) {
+                nmsItemStack = net.minecraft.world.item.ItemStack.NULL_ITEM;
+            }
         } else {
             nmsItemStack = CraftItemStack.asNMSCopy(itemStack);
         }
